x-pack/apm-server/sampling/processor.go

```diff
@@ -164,7 +170,6 @@ func (p *Processor) ProcessBatch(ctx context.Context, batch *model.Batch) error
 			events = events[:n-1]
 			i--
 		}
-		p.updateProcessorMetrics(report, stored)
```

Comment:
I think you uncovered a more serious bug. IIANM, ProcessBatch is dropping metrics. I think we should leave this call to updateProcessorMetrics here, and update the switch to just skip over non transaction/span events.